examples/lib/stories/experimental/loading_screen.dart

```diff
@@ -0,0 +1,239 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/widgets.dart';
+import 'package:flutter/material.dart' hide Image, Draggable;
+
+class LoadingScreenExample extends FlameGame {
+  LoadingScreenExample();
+
+  static const description = '''
+  Just enjoy how the progress bar fills and numbers grows...
+  The loading screen and in-game components are using the same API and same 
+  message stream.
+  ''';
+
+  final _progressNotifier = GameLoadProgressNotifier<ProgressMessage>();
+
+  /// Override [progressNotifier] getter with specifying target message type.
+  /// Is necessary to avoid [dynamic] types.
+  @override
+  GameLoadProgressNotifier<ProgressMessage> get progressNotifier =>
+      _progressNotifier;
+
+  double anyProgressEmulation = 0;
+  int messagesSent = 0;
+
+  @override
+  FutureOr<void> onLoad() async {
+    /// the message will be send to loading screen, if the one is specified
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(message: 'onLoad started', progress: 1),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: 'Something happen while loading...',
+        progress: 50,
+      ),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: '...almost done!..',
+        progress: 90,
+      ),
+    );
+
+    add(
+      TextComponent(text: 'This is in-game text components')
+        ..anchor = Anchor.topCenter
+        ..x = size.x / 2
+        ..y = 50,
+    );
+
+    /// These components will receive notifications during game loop.
+    add(InGameProgressText(verticalPosition: 100, size: size));
+    add(InGameProgressText(verticalPosition: 150, size: size));
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: 'Finished! Be ready to play!',
+        progress: 100,
+      ),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    /// You should to decide, what type of message would be trigger for
+    /// showing game widget. You should to sent such message manually and
+    /// also manually build game widget in loading screen component.
+    /// See [_LoadingScreenExampleWidgetState.build]
+    progressNotifier.reportLoadingProgress(const ProgressMessage.finished());
+
+    return super.onLoad();
+  }
+
+  /// This is just an example of sending messages to game components.
+  @override
+  void update(double dt) {
+    anyProgressEmulation += dt;
+    final probability = Random().nextInt(100);
+    if (probability < 1) {
+      messagesSent++;
+      progressNotifier.reportLoadingProgress(
```

Comment:
What is the best way to separate component and game `progressNotifier`s?
let's say i want to have a 2 progress systems.

1. Game Loading progress (will be used by Flutter)
2. Levels loading progress (will be used by Components)

What is the preferred way to implement this? I don't want to receive component messages from Flutter and game load progress from within Components.

Reply:
I did not understand clearly your point. If you have two progress systems - well, ok, it will work. 
If the game is still loading, there are no messages from the component's lifecycle, because the lifecycle still has not started. 
If the game is already loaded and you have a message from the component - you could just ignore it in your loading widget, setting a variable to track the difference between states, for example... 
I see I missed it in my example, and every future message triggers `setState` - will fix it.